baconTests/model/TransactionTests.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  TransactionTests.swift
+//  baconTests
+//
+//  Created by Fabian Terh on 19/3/19.
+//  Copyright © 2019 nus.CS3217. All rights reserved.
+//
+
+import XCTest
+@testable import bacon
+
+class TransactionTests: XCTestCase {
+    let testDate = Date()
+    let testFrequency = try! TransactionFrequency(nature: .oneTime)
+
+    func test_init_validInput_success() {
+        let transaction = try! Transaction(date: testDate,
+                                           type: .expenditure,
+                                           frequency: testFrequency,
+                                           category: .bills,
+                                           amount: 1)
+        XCTAssertEqual(transaction.date, testDate)
+        XCTAssertEqual(transaction.type, .expenditure)
+        XCTAssertEqual(transaction.frequency, testFrequency)
+        XCTAssertEqual(transaction.category, .bills)
+        XCTAssertEqual(transaction.amount, 1)
+        XCTAssertEqual(transaction.description, "")
+    }
+
+    func test_init_invalidInput() {
+        XCTAssertThrowsError(try Transaction(date: testDate,
+                                             type: .expenditure,
+                                             frequency: testFrequency,
+                                             category: .bills,
+                                             amount: -1)) { (err) in
```

Comment:
what about 0 amount? Is it valid?

Reply:
Should it be an error if it is 0? Then we prompt the user for input, if they attempt to create a transaction with 0 amount.

Reply:
Yup, it should fail (throw an error). I've added a new test case to reflect that.